src/app/rpmostree-builtin-status.c

```diff
@@ -21,301 +22,204 @@
-               max_id_len+buffer, HEADER_ID,
-               max_osname_len+buffer, HEADER_OSNAME,
-               max_refspec_len, HEADER_REFSPEC);
-    }
-  /* header for "pretty" row output */
-  else
-    printchar ("=", 60);
-
-  n = deployment_dicts->len;
-
-  /* print entries for each deployment */
-  for (i = 0; i < n; i++)
-    {
-      GVariantDict *dict;
-      g_autoptr(GDateTime) timestamp = NULL;
-      g_autofree char *timestamp_string = NULL;
-      g_autofree gchar *truncated_csum = NULL;
-      g_autoptr(GVariant) signatures = NULL;
-
-      gchar *id = NULL; /* borrowed */
-      gchar *origin_refspec = NULL; /* borrowed */
-      gchar **origin_packages = NULL; /* borrowed */
-      gchar *os_name = NULL; /* borrowed */
-      gchar *version_string = NULL; /* borrowed */
-      gchar *checksum = NULL; /* borrowed */
-      g_autofree gchar *origin_refspec_description = NULL;
-
-      guint64 t = 0;
-      gint serial;
-      gboolean is_booted = FALSE;
-
-      dict = g_ptr_array_index (deployment_dicts, i);
-
-      g_variant_dict_lookup (dict, "id", "&s", &id);
-      g_variant_dict_lookup (dict, "osname", "&s", &os_name);
-      g_variant_dict_lookup (dict, "serial", "i", &serial);
-      g_variant_dict_lookup (dict, "checksum", "s", &checksum);
-      g_variant_dict_lookup (dict, "version", "s", &version_string);
-      g_variant_dict_lookup (dict, "timestamp", "t", &t);
-      g_variant_dict_lookup (dict, "origin", "s", &origin_refspec);
-      g_variant_dict_lookup (dict, "packages", "^a&s", &origin_packages);
       signatures = g_variant_dict_lookup_value (dict, "signatures",
                                                 G_VARIANT_TYPE ("av"));
 
-      if (origin_packages)
-        {
-          g_autofree gchar *origin_packages_plus =
-            format_layered_packages_plus (origin_packages);
-          origin_refspec_description = g_strconcat (origin_refspec, origin_packages_plus, NULL);
-        }
+      if (first)
+        first = FALSE;
       else
-        {
-          origin_refspec_description = g_strdup (origin_refspec);
-        }
+        g_print ("\n");
 
       is_booted = g_strcmp0 (booted_id, id) == 0;
 
-      timestamp = g_date_time_new_from_unix_utc (t);
-      if (timestamp != NULL)
-        timestamp_string = g_date_time_format (timestamp, "%Y-%m-%d %T");
-      else
-        timestamp_string = g_strdup_printf ("(invalid)");
-
-      /* truncate checksum */
-      truncated_csum = g_strndup (checksum, CSUM_DISP_LEN);
+      g_print ("%s ", is_booted ? libsd_special_glyph (BLACK_CIRCLE) : " ");
 
-      /* print deployment info column */
-      if (!opt_pretty)
+      if (origin_refspec)
+        g_print ("%s", origin_refspec);
+      else
+        g_print ("%s", checksum);
+      g_print ("\n");
+      if (version_string)
+        print_kv ("Version", max_key_len, version_string);
+      print_kv (origin_packages ? "BaseCommit" : "Commit", max_key_len, checksum);
```

Comment:
I see a fundamental distinction between *remote* commits and local commits.  A remote commit is something that came from "someone else" possibly.  It will have errata and changelogs etc.  You'll be able to find the checksum externally.  None of that is true for the local commit.

Now we could try to be a bit more accurate and detect whether we're tracking a refspec with a remote as origin?  But does the core distinction make sense to you?